### PR TITLE
[DO NOT MERGE] Initial commit to add workload management apis to synapse RP

### PR DESCRIPTION
**Pull request by tunafish88**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by raych1** [comment]
Hi @tunafish88 , some api path doesn't have the MIME type definition.  It would be good to add them to root level instead of api path level if they're same.
```
"consumes": [
    "application/json"
  ],
  "produces": [
    "application/json"
  ]`
```

**Comment by tunafish88** [comment]
> Hi @tunafish88 , some api path doesn't have the MIME type definition. It would be good to add them to root level instead of api path level if they're same.
> 
> ```
> "consumes": [
>     "application/json"
>   ],
>   "produces": [
>     "application/json"
>   ]`
> ```

Thanks. This helped

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (ArcturusZhang) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3008]: Could we sort these response code so that the default response is the last one?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (ArcturusZhang) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3072]: Could we sort these response code so that the default response is the last one?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (ArcturusZhang) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3266]: Could we sort these response code so that the default response is the last one?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (ArcturusZhang) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3337]: Could we sort these response code so that the default response is the last one?

**Review by ArcturusZhang — COMMENTED** [review]
Some really minor comments

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json]: Sure.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json]: Sure

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json]: Sure

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json]: sure

**Comment by ArcturusZhang** [comment]
Hi @tunafish88 this PR now looks good to me.

Since this PR still needs ARM's feedback, I will hold my approval to avoid any accidentally merging. I will approve and merge this once ARM signs off on this PR.

Thanks

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5008]: can this be a string enum?  How does the user know what values are accepted here?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5012]: Please specify what units this uses in the property name and/or description

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5073]: "format": "datetime" for startTime and endTime

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5081]: same string enum comment as the other importance property

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5008]: Hi, The Synapse RP API for which this documentation is being generated is like a proxy level that calls into SQL API that already exists today and is in use. 
So changing the SQL API  to change this property to an enum is a bigger change. 
Is it ok if we describe the importance levels like this:
There are five levels of importance: low, below_normal, normal, above_normal, and high. (from here : https://docs.microsoft.com/en-us/azure/synapse-analytics/sql-data-warehouse/sql-data-warehouse-workload-importance )

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5073]: Will fix this.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5012]: Will change the description to "description": "The workload group query execution timeout in seconds.",

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5081]: There are five levels of importance: low, below_normal, normal, above_normal, and high.

**Review by pilor — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
Hi @tunafish88 could you please also update the examples to fix the ModelValidation check?

**Comment by tunafish88** [comment]
Closing this pull request because after making ARM recommended changes to treat some strings representing hh:mm to format 'date-time' , the API behavior is broken. The API only accepts string , and takes 'hh:mm'. For now, since we do not need this swagger urgently, I'll be closing this PR and we don't need the check in right now. We will discuss with the internal team and send out a new PR if needed.

**Pull request closed without merging**
